src/Mono.Android/Test/Android.Runtime/InputStreamInvokerTest.cs

```diff
@@ -0,0 +1,63 @@
+                        Java.IO.FileInputStream fileStream = new Java.IO.FileInputStream (path);
+                        Java.IO.ByteArrayInputStream byteStream = new Java.IO.ByteArrayInputStream (Encoding.ASCII.GetBytes (text));
+
+                        using (var stream = new InputStreamInvoker (fileStream)) {
+                                byte[] bytes = new byte[text.Length];
+                                Assert.IsTrue (stream.CanRead);
+                                Assert.IsTrue (stream.CanSeek);
+                                Assert.IsFalse (stream.CanTimeout);
+                                Assert.IsFalse (stream.CanWrite);
+                                Assert.AreEqual (stream.Length, 24);
+                                Assert.AreEqual (stream.Seek (8, SeekOrigin.Begin), 8);
+                                Assert.AreEqual (stream.Position, 8);
+                                Assert.AreEqual (stream.Read (bytes, 0, 7), 7);
+                                Assert.AreEqual ("so good", Encoding.ASCII.GetString (bytes, 0, 7));
+                                Assert.AreEqual (stream.Seek (1, SeekOrigin.Current), 16);
+                                Assert.AreEqual (stream.Read (bytes, 0, 7), 7);
+                                Assert.AreEqual ("looking", Encoding.ASCII.GetString (bytes, 0, 7));
+                                Assert.AreEqual (stream.Seek (-text.Length, SeekOrigin.End), 0);
+                                Assert.AreEqual (stream.Read (bytes, 0, 7), 7);
+                                Assert.AreEqual ("Your're", Encoding.ASCII.GetString (bytes, 0, 7));
+                                stream.Position = text.Length - 1;
+                                Assert.AreEqual (stream.Read (bytes, 0, 1), 1);
+                                Assert.AreEqual ("!", Encoding.ASCII.GetString (bytes, 0, 1));
+                        }
+
+                        using (var stream = new InputStreamInvoker (byteStream)) {
+                                byte[] bytes = new byte[text.Length];
+                                Assert.IsTrue (stream.CanRead);
+                                Assert.IsFalse (stream.CanSeek);
+                                Assert.IsFalse (stream.CanTimeout);
+                                Assert.IsFalse (stream.CanWrite);
+                                Assert.Throws<NotSupportedException> (() => { var _ = stream.Length; });
+                                Assert.Throws<NotSupportedException> (() => { var _ = stream.Position; });
+                                Assert.Throws<NotSupportedException> (() => { stream.Position = 1; });
+                                Assert.Throws<NotSupportedException> (() => { stream.Seek (1, SeekOrigin.Begin); });
+                                Assert.AreEqual (stream.Read (bytes, 0, text.Length), text.Length);
+                                Assert.AreEqual (text, Encoding.ASCII.GetString (bytes, 0, 1));
```

Comment:
This appears to be the assert which is failing, and upon review it doesn't make sense to me: `Encoding.ASCII.GetBytes(bytes, 0, 1)` will only return a string with a single character, while `text` contains 24 characters.

Should this instead be?

```csharp
Assert.AreEqual (text, Encoding.ASCII.GetString (bytes, 0, text.Length));
```

Reply:
Sorry about that, yes looks like I copied that last line without running it. text.Length instead of 1 is correct.